### PR TITLE
feat: enable generic transaction batching

### PR DESCRIPTION
**Pull request by mpetrun5**
Remove custom deposit event listener and enable proposal batching

## Related Issue Or Context

Closes: #216 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in docs.
- [ ] I have added tests to cover my changes.
- [ ] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

**Review by tcar121293 — APPROVED** [review]

File: chains/evm/executor/executor.go [hunk 1]
```diff
@@ -27,6 +27,12 @@ import (
 )
 
 const TRANSFER_GAS_COST = 200000
+const MAX_TRANSACTION_COST = 25000000
```
Comment (MakMuftic) [inline chains/evm/executor/executor.go:30]: 25 million? That doesn't sound like the correct number.

**Review by nmlinaric — APPROVED** [review]

File: chains/evm/executor/executor.go [hunk 2]
```diff
@@ -180,23 +180,61 @@ func (e *Executor) watchExecution(ctx context.Context, cancelExecution context.C
...
+			return nil, err
+		}
+
+		evmProposal := chains.NewProposal(prop.Source, prop.Destination, prop.DepositNonce, prop.ResourceId, prop.Data, prop.Metadata)
+		isExecuted, err := e.bridge.IsProposalExecuted(evmProposal)
+		if err != nil {
+			return nil, err
+		}
+		if isExecuted {
+			log.Info().Msgf("Prop %p already executed", prop)
```
Comment (MakMuftic) [inline chains/evm/executor/executor.go:203]: I know it was like this before, but I don't see reason why not use full word in logs
```suggestion
			log.Info().Msgf("Proposal %p already executed", prop)
```

File: chains/evm/executor/executor.go (hunk 1 above)
Comment (mpetrun5) [reply chains/evm/executor/executor.go:30]: Current max block size on Ethereum is 30 million.
I reduced to be a bit less.
Though now that you mention it maybe makes sense to make configurable if some EVM has a different block size (for sure it has)

Comment (MakMuftic) [reply chains/evm/executor/executor.go:30]: It actually makes sense now, but you are right having this as a configurable property per chain probably makes the most sense (25mil can be the default value).

Comment (mpetrun5) [reply chains/evm/executor/executor.go]: I used the `gasLimit` field already in the configuration because we don't use that anymore and it made sense

**Review by MakMuftic — APPROVED** [review]

**Pull request merged**